rllib/core/models/tests/test_cnn_encoders.py

```diff
@@ -0,0 +1,130 @@
+import unittest
+import itertools
+
+from ray.rllib.core.models.base import ENCODER_OUT, STATE_IN, STATE_OUT
+from ray.rllib.core.models.configs import CNNEncoderConfig
+from ray.rllib.models.utils import get_filter_config
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.test_utils import framework_iterator
+
+_, tf, _ = try_import_tf()
+torch, _ = try_import_torch()
+
+
+class TestCNNEncoders(unittest.TestCase):
+    def test_cnn_encoders(self):
+        """Tests building CNN encoders properly and checks for correct architecture."""
+
+        # Input dims supported by RLlib via get_filter_config().
+        inputs_dimss = [
+            [480, 640, 3],
+            [480, 640, 1],
+            [240, 320, 3],
+            [240, 320, 1],
+            [96, 96, 3],
+            [96, 96, 1],
+            [84, 84, 3],
+            [84, 84, 1],
+            [64, 64, 3],
+            [64, 64, 1],
+            [42, 42, 3],
+            [42, 42, 1],
+            [10, 10, 3],
+        ]
+
+        cnn_activations = [None, "linear", "relu"]
+
+        cnn_use_layernorms = [False, True]
+
+        output_dimss = [[1], [100]]
+
+        output_activations = cnn_activations
+
+        use_biases = [False, True]
+
+        for permutation in itertools.product(
+            inputs_dimss,
+            cnn_activations,
+            cnn_use_layernorms,
+            output_activations,
+            output_dimss,
+            use_biases,
+        ):
+            (
+                inputs_dims,
+                cnn_activation,
+                cnn_use_layernorm,
+                output_activation,
+                output_dims,
+                use_bias,
+            ) = permutation
+
+            filter_specifiers = get_filter_config(inputs_dims)
+
+            print(
+                f"Testing ...\n"
+                f"inputs_dims: {inputs_dims}\n"
+                f"cnn_filter_specifiers: {filter_specifiers}\n"
+                f"cnn_activation: {cnn_activation}\n"
+                f"cnn_use_layernorm: {cnn_use_layernorm}\n"
+                f"output_activation: {output_activation}\n"
+                f"output_dims: {output_dims}\n"
+                f"use_bias: {use_bias}\n"
+            )
+
+            config = CNNEncoderConfig(
+                input_dims=inputs_dims,
+                cnn_filter_specifiers=filter_specifiers,
+                cnn_activation=cnn_activation,
+                cnn_use_layernorm=cnn_use_layernorm,
+                output_dims=output_dims,
+                output_activation=output_activation,
+                use_bias=use_bias,
+            )
+
+            # To compare number of params between frameworks.
+            tf_counts = None
```

Comment:
initialize both tf and torch counts in a unified and general way.

```
parmeter_count_dict = {}

for fw in ...:
     parmeter_count_dict[fw] = model.get_num_parameters()
     
     if len(parameter_count_dict) > 1:
           ref_count = next(iter(parameter_count_dict.values()))
           # check the trainable parameters
           self.assertTrue(all(ref_count[0] == v[0] for k, v in parameter_count_dict))
           # check the non-trainable parameters
           self.assertTrue(all(ref_count[1] == v[1] for k, v in parameter_count_dict))
``` 

This way, later, if only add one more framework this test will automatically work. Generally the correctness of your test should not depend on the order of iteration over your framework.